src/UartSerial.cpp

```diff
@@ -130,10 +136,43 @@ int UartSerial::read()
 
 void UartSerial::write(uint8_t byte_)
 {
-  ::write(_fd, &byte_, 1);
+  if(_fd != -1)
+    ::write(_fd, &byte_, 1);
 }
 
-void UartSerial::registerSerialEventCallback (serialEvent bytes_available_)
+void UartSerial::registerSerialEventCallback (serialEvent bytes_available_, void *context_)
 {
-  _bytes_available_callback = bytes_available_;
+  if(_fd == -1) return;
+
+  // For now only allow a single callback to be assigned
+  if(_bytes_available_callback == nullptr)
+  {
+    _bytes_available_callback = bytes_available_;
+    _callbackContext = context_;
+
+    // Setup poll struct
+    _fds[0].fd = _fd;
+    _fds[0].events = POLLIN;
+
+    _polling = true;
```

Comment:
Okay, I totally understand why this is here, but I think the thread *Line 164.5* needs to be gated on `UartSerial::begin`. Since this is the Linux implementation and you're already using `std::thread`, I would probably take it to the next level with a `std::future` and `std::promise`.